# Import and Tidy

## Import csv data

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/data/jandrewtorres/data.csv')

## Check columns and dtypes of columns

In [3]:
df.dtypes

number_people             int64
date                     object
timestamp                 int64
day_of_week               int64
is_weekend                int64
is_holiday                int64
temperature             float64
is_start_of_semester      int64
is_during_semester        int64
month                     int64
hour                      int64
dtype: object

In [13]:
df = df.drop(['date','timestamp'], axis=1)

## Look at description of dataset to determine what needs to be cleaned

In [14]:
df.describe()

number_people   day_of_week    is_weekend    is_holiday   temperature  \
count   62184.000000  62184.000000  62184.000000  62184.000000  62184.000000   
mean       29.072543      2.982504      0.282870      0.002573     58.557108   
std        22.689026      1.996825      0.450398      0.050660      6.316396   
min         0.000000      0.000000      0.000000      0.000000     38.140000   
25%         9.000000      1.000000      0.000000      0.000000     55.000000   
50%        28.000000      3.000000      0.000000      0.000000     58.340000   
75%        43.000000      5.000000      1.000000      0.000000     62.280000   
max       145.000000      6.000000      1.000000      1.000000     87.170000   

       is_start_of_semester  is_during_semester         month          hour  
count          62184.000000        62184.000000  62184.000000  62184.000000  
mean               0.078831            0.660218      7.439824     12.236460  
std                0.269476            0.473639      3.445069      6.717631  
min                0.000000            0.000000      1.000000      0.000000  
25%                0.000000            0.000000      5.000000      7.000000  
50%                0.000000            1.000000      8.000000     12.000000  
75%                0.000000            1.000000     10.000000     18.000000  
max                1.000000            1.000000     12.000000     23.000000

We can see that the minimum and maximum values are reasonable:
- number_people [0, 145]
- day_of_week [0, 6]
- is_weekend [0, 1]
- is_holiday [0, 1]
- temperature[38.14, 87.17]
- is_start_of_semester [ 0, 1]
- is_during_semester [0, 1]
- month[1, 12]
- hour [0, 23]

## Export tidied data

In [16]:
df.to_csv('/data/jandrewtorres/tidied_data.csv', index=False)